src/CentrifugeRouter.sol

```diff
@@ -9,27 +9,23 @@ import {IERC7540Vault} from "src/interfaces/IERC7540.sol";
 import {ICentrifugeRouter} from "src/interfaces/ICentrifugeRouter.sol";
 import {IPoolManager} from "src/interfaces/IPoolManager.sol";
 import {IEscrow} from "src/interfaces/IEscrow.sol";
-
-interface GatewayLike {
-    function topUp() external payable;
-    function estimate(bytes calldata payload) external returns (uint256[] memory tranches, uint256 total);
-}
+import {IGateway} from "src/interfaces/gateway/IGateway.sol";
 
 contract CentrifugeRouter is Auth, ICentrifugeRouter {
     IEscrow public immutable escrow;
+    IGateway public immutable gateway;
     IPoolManager public immutable poolManager;
-    GatewayLike public immutable gateway;
 
     address constant UNSET_INITIATOR = address(1);
     address internal _initiator = UNSET_INITIATOR;
 
     /// @inheritdoc ICentrifugeRouter
     mapping(address controller => mapping(address vault => uint256 amount)) public lockedRequests;
 
-    constructor(address escrow_, address poolManager_, address gateway_) {
+    constructor(address escrow_, address gateway_, address poolManager_) {
         escrow = IEscrow(escrow_);
+        gateway = IGateway(gateway_);
```

Comment:
Any particular reason why we switched from "in-contract" `GatewayLike` to full-blown `IGateway` ?

Using the `GatewayLike` vs `import IGateway` approach
Pros:
 - Not all code that is defined in the interface gets copied into the file that imports the interface. - Grows deployment size.
 - Expose only functions that are meant to be called within the scope of the contract that's using it.  - Might result in calling something that is unexpected.
 
 Cons: 
 - Every single time you change a signature, you have to go through all `GatewayLike` definitions and change the signature there as well which is time consuming and annoying.